### PR TITLE
Improve README for project onboarding (#3936)

### LINKED ISSUE #3109: Break README.md out into smaller wiki files
### Description
Our readme file has ballooned in size and scope and could use some reorganization.

### Proposed solution 
Make the readme a more generic project description, with links to some of our various environments.

Reorganize the content of the existing readme in a "getting started" wiki (we'll want to link this page on the new readme).

### Additional context
The wiki is much easier to update than the readme as it doesn't require a PR, which is one of the reasons we'd like to make this change.

### PR DESCRIPTION
**Pull request by emyl3**
# PULL REQUEST

## Related Issue
Part of cleanup effort for #3936 
Resolves #3109

## Changes Proposed
- moves sections in README to separate wiki pages
- cleans up and updates the README to provide the following:
  - context for what SimpleReport is
  - how to get started
  - how to contribute

## Additional Information
- work is still being done to clean up/merge these two pages:
  - https://github.com/CDCgov/prime-simplereport/wiki/Running-outside-of-docker
  - https://github.com/CDCgov/prime-simplereport/wiki/WIP:-Local-development-setup-(outside-Docker-&-IntelliJ)

## Testing
- Check the links to see they work and README content makes sense

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,41 +33,9 @@ The Simple Report application stack consists of:
...
-"The build failed because the process exited too early. This probably means the system ran out of memory or someone called `kill -9` on the process."
-```
-
-How to update resources limits on your [Mac](https://docs.docker.com/desktop/mac/#resources), and [Windows](https://docs.docker.com/desktop/windows/#resources) machines.
-
-## Running outside of docker
-
-If you wish to run outside of docker compose (bare metal), you'll need to install all the dependencies locally.
-See more [info here](https://github.com/CDCgov/prime-simplereport/wiki/Running-outside-of-docker).
+There are two different ways to run the app for local development
```
Comment (emmastephenson) [inline README.md:36]: Could you leave a note in here that most feature-facing devs use the bare metal setup, but DevOps generally prefers Docker?

File: README.md [hunk 2]
```diff
@@ -34,41 +33,9 @@ The Simple Report application stack consists of:
...
+1. [Docker and docker compose](https://github.com/CDCgov/prime-simplereport/wiki/%5BSetup%5D-Docker-and-docker-compose-development)
+2. [Bare metal (outside of Docker)](https://github.com/CDCgov/prime-simplereport/wiki/%5BSetup%5D-Running-outside-of-docker)
```
Comment (emmastephenson) [inline README.md:38]: Not really part of this PR, but I think this wiki page has a little too much data - it also includes information about rollbacks, database cleaning, etc that is already on this README. If you could break those sections into separate wiki pages it'd be great, thank you!

**Review by emmastephenson — COMMENTED** [review]
Thanks Elisa!

File: README.md (hunk 2 above)
Comment (emyl3) [reply README.md:38]: Yes! It's on my todo list hahah Thank you for flagging this! 😸

File: README.md (hunk 1 above)
Comment (emyl3) [reply README.md:36]: Yes! Can do!

Comment (emyl3) [reply README.md:36]: Done!

**Review by emmastephenson — APPROVED** [review]
Looks great, thanks Elisa!

**Comment by emyl3** [comment]
Changed the title to DO NOT REVIEW because I realized I should just clean up all the README changes all at once 😅

File: README.md [hunk 3]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-  - [Frontend](#frontend)
-    - [Linters](#linters)
-    - [Storybook and Chromatic](#storybook-and-chromatic)
-  - [PR Conventions](#pr-conventions)
-  - [Cloud Environments](#cloud-environments)
-  - [Cloud Architecture](#cloud-architecture)
-  - [Deploy](#deploy)
-    - [Revert to a Previous Release](#revert-to-a-previous-release)
-    - [Deploy With Action](#deploy-with-action)
+## Overview
```
Comment (emyl3) [inline README.md:12]: ⚠️ Got the content for the "Overview" section from our product vision doc https://docs.google.com/document/d/1Fk7mu7ECsxiVZ_6W1KJcQ_dBIfo46dTu4GdBrbaZGDY/edit

Specifically the "Vision" and "Mission" sections

File: README.md [hunk 4]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
+SimpleReport, as part of PRIME (Pandemic-Ready Interoperability Modernization Effort), aims to improve data quality, completeness and timeliness of public health case reporting for COVID-19, the flu and future infectious diseases.
 
-## Developing locally
+SimpleReport does this by creating a real-time avenue for providers who do not have traditional healthcare IT systems to send high quality, timely, structured data to public health. This ensures that public health has a clear(er) and more timely, complete picture of COVID positivity and can take appropriate steps to intervene and prevent further community spread.
 
-The Simple Report application stack consists of:
-- PostgreSQL database
-- Java Spring Boot backend
-- React frontend
+## Getting Started
```
Comment (emyl3) [inline README.md:17]: ⚠️ Please feel free to edit this wiki page with links to other wiki pages that would be helpful for a new dev to SR!

File: README.md [hunk 5]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
+To start developing for SimpleReport, refer to our [Getting Started](https://github.com/CDCgov/prime-simplereport/wiki/Getting-Started) wiki page.
 
-For ease of local setup, we have created a docker compose setup which builds docker images for all the necessary components. It also uses nginx and mkcert to host the app locally over HTTPS.
```
Comment (emyl3) [inline README.md:37]: ⚠️ moved to here https://github.com/CDCgov/prime-simplereport/wiki/Docker-and-docker-compose-development

File: README.md [hunk 6]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-
-#### Notes
-Make sure docker has access to enough resources on your local machine. You'll know it doesn't if you get an error from any of your containers similar to this:
-```
-"The build failed because the process exited too early. This probably means the system ran out of memory or someone called `kill -9` on the process."
-```
-
-How to update resources limits on your [Mac](https://docs.docker.com/desktop/mac/#resources), and [Windows](https://docs.docker.com/desktop/windows/#resources) machines.
-
-## Running outside of docker
```
Comment (emyl3) [inline README.md:68]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Running-outside-of-docker

File: README.md [hunk 7]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-
-If you wish to run outside of docker compose (bare metal), you'll need to install all the dependencies locally.
-See more [info here](https://github.com/CDCgov/prime-simplereport/wiki/Running-outside-of-docker).
-
-## Roles
```
Comment (emyl3) [inline README.md:73]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/User-roles

File: README.md [hunk 8]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-
-These roles are controlled via Okta groups.
-
-## Support admin
-
-You can make your user a support admin by assigning yourself the `SR-DEV-ADMINS` group in Okta Preview.
-
-Support admins can access the `/admin` paths and support admin APIs.
-
-## Database cleaning
```
Comment (emyl3) [inline README.md:91]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Database-maintenance

File: README.md [hunk 9]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-
-To roll back to a certain tag:
-
-```
-docker compose run --rm backend gradle liquibaseRollback -PliquibaseCommandValue=${TAG}
-```
-
-If you are required to roll back a non-local database, you may generate the required SQL to execute elsewhere. Use `liquibaseRollbackToDateSQL` or `liquibaseRollbackCountSQL` in the manner described above to write the rollback SQL to stdout.
-
-## API testing
```
Comment (emyl3) [inline README.md:125]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/API-Testing-with-Insomnia

File: README.md [hunk 10]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-
-You can make requests to the GraphQL API using [Insomnia](https://insomnia.rest/download) or a similar client. You would need to point the api endpoint to the backend at: `https://localhost.simplereport.gov/api/graphql` This gives you a preview to query/mutate the local database.  
-  
-When using Insomnia, you'll also need to pass your access_token in the `Authorization` header - this can be found in Application -> Local Storage -> Access Token on a Chrome browser, or by typing `localStorage.getItem("access_token")` into the console. Copy that value, then create an `Authorization` header with the value `Bearer ${access_token}`.
-
-## Tests
```
Comment (emyl3) [inline README.md:131]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Running-tests

File: README.md [hunk 11]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-yarn test:frontend
-```
-
-Run backend unit tests:
-
-```
-yarn test:backend
-```
-
-## E2E tests
```
Comment (emyl3) [inline README.md:145]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/E2E-tests

File: README.md [hunk 12]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-  - Prerequisite(s):
-    - the app must be running locally with the frontend base URL set to `http://localhost:3000` and backend running with the _dev_ profile
-    - `REACT_APP_OKTA_ENABLED=false` in `frontend/.env.local`
-- `yarn e2e:local:okta`
-  - Same as `yarn e2e:local`, except that specs with a login step will log in to Okta. This is more similar to how the tests run in our GitHub PR workflow.
-  - Prerequisite(s):
-    - the app must be running [locally with Okta](https://github.com/CDCgov/prime-simplereport/wiki/Running-outside-of-docker#running-locally-with-okta) with the frontend base URL set to `http://localhost:3000` and the backend running with the _okta-local_ and _local_ profiles
-    - `REACT_APP_OKTA_ENABLED=true` in `frontend/.env.local`
-
-<!-- Dockerized end-to-end tests are currently broken
```
Comment (emyl3) [inline README.md:192]: ⚠️ Remains commented out on the E2E Tests wiki page 😅

File: README.md [hunk 13]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-- our tests expect there to only be 1 facility and will fail if there are multiple facilities for whatever org the test user is authenticated into
-  - when running the tests with Okta, the test user Cypress McTestUser has admin access to an org with external ID `DAT_ORG`, which with our default sample data will have 2 facilities
-- `00-health_check.js` is used in our CI to check that the app is up but fails when running locally with either of the commands listed here
-- `01-organization_sign_up_spec.js` is not re-runnable
-- `05-self_registration_spec.js` expects the org your test user is in to have a patient registration link generated
-
-See the [Cypress documentation](https://docs.cypress.io/api/table-of-contents) for writing new tests. If you need to generate new Wiremock mappings for external services, see [this wiki page](https://github.com/CDCgov/prime-simplereport/wiki/WireMock).
-
-
-## Load Tests
```
Comment (emyl3) [inline README.md:231]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Running-tests

File: README.md [hunk 14]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-SimpleReport leverages Locust for running its load tests. Currently, Locust starts as part of the local development build, and spawns three separate worker containers for inundating the system.
-
-To compose Locust with the rest of the development stack, run `yarn locust` instead of `yarn start`. *(NOTE: All other prerequisites listed above in "Developing locally" are still required for tests to function.)*
-
-To access the Locust user interface, navigate to `http://localhost:8089/`. Load tests will not kick off automatically; they must be manually invoked.
-
-Tests can be configured by modifying `backend/locust/locustfile.py`.
-
-
-## Twilio
```
Comment (emyl3) [inline README.md:241]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Twilio

File: README.md [hunk 15]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-
-```
-twilio:
-  enabled: true
-  from-number: +13214560987
-```
-
-These can also be set by environment variable if desired.
-
-## MailHog
```
Comment (emyl3) [inline README.md:253]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/MailHog

File: README.md [hunk 16]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-
-MailHog is an email-testing tool with a fake SMTP server underneath, we can use it to test sending emails locally.
-
-- Mailhog client runs on docker, `docker compose up -d mailhog`
-
-- Access mailhog inbox on `http://localhost:8025/`
-
-## Frontend
```
Comment (emyl3) [inline README.md:261]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Frontend-Overview

File: README.md [hunk 17]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-
-The front end is a React app. The app uses [Apollo](https://www.apollographql.com/) to manage the graphql API. For styling the app leverages the [U.S. Web Design System (USWDS)](https://designsystem.digital.gov/). Ensure your node version is version 14 in order for
-yarn to build packages correctly.
-
-### Linters
```
Comment (emyl3) [inline README.md:266]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/SimpleReport-Style-Guide as a new section

File: README.md [hunk 18]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-GitHub Actions is configured to run these linters on every pull request, so you must resolve all mismatches/errors prior to merging.
-There are a few ways to manage this:
-
-1. Run `yarn lint:write` in the `frontend/` dir, and `./gradlew spotlessApply` in the `backend/` dir, before every commit
-1. Enable the optional pre-commit hook by running `yarn lefthook install` in the root dir (install lefthook if you haven't already done so during setup by running `yarn install`)
-1. Add extensions to your code editor that runs the linters for you on save, e.g. [prettier-vscode](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode), [vscode-eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), [vscode-google-java-format](https://marketplace.visualstudio.com/items?itemName=ilkka.google-java-format)
-
-You can also run the pre-commit hooks in docker using `yarn lefthook run in-docker`.
-
-### Storybook and Chromatic
```
Comment (emyl3) [inline README.md:279]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Storybook-and-Chromatic

File: README.md [hunk 19]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-[Chromatic](https://www.chromatic.com/) is a web-based tool for Storybook that helps speed UI
-component development. It provides regression testing and review. It also allows for publication
-of the Storybook.
-
-Changes to the Storybook are sent to Chromatic when changes to the frontend source are push to a
-any branch. The changes are automatically accepted on merge to `main`.
-
-View the [SimpleReport Storybook](https://main--60a556a7c807cc0039ec6786.chromatic.com/)
-
-## PR Conventions
```
Comment (emyl3) [inline README.md:297]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/PR-Conventions

File: README.md [hunk 20]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-
-The convention for branch naming on this project is {firstName}/{ticketNumber}-{short-description}. This makes it easier for other developers to find your changes.
-
-If you're merging large or complex changes, it is strongly recommended that you smoke test them in `dev`, `test`, or `pentest`. These three environments are roughly the same, with small configuration changes between each (`test` sends text results for pxp while `dev` and `pentest` do not, for example.)
-
-For all changes, please ensure the PR checklist is completed before sending out for review. If you're making UI changes, make sure to screenshot and get approval from a designer or product manager, in addition to engineers.
-
-We require two reviewers per changeset, and you cannot merge until all commits have been reviewed.
-
-## Cloud Environments
```
Comment (emyl3) [inline README.md:307]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Cloud-Environments as a new section

File: README.md [hunk 21]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-|   dev4   |   [/app/static/commit.txt](https://dev4.simplereport.gov/app/static/commit.txt)   |   [/api/actuator/info](https://dev4.simplereport.gov/api/actuator/info)   |     [Action](#deploy-with-action)     |           Development testing/PR validation before merging to `main`           |
-|   test   |   [/app/static/commit.txt](https://test.simplereport.gov/app/static/commit.txt)   |   [/api/actuator/info](https://test.simplereport.gov/api/actuator/info)   |            Push to `main`             |                   Automated testing agsinst `main` (e2e/load)                  |
-|  pentest | [/app/static/commit.txt](https://pentest.simplereport.gov/app/static/commit.txt)  | [/api/actuator/info](https://pentest.simplereport.gov/api/actuator/info)  |     [Action](#deploy-with-action)     |                   Penetration testing (no alerting enabled)                    |
-
-## Cloud Architecture
```
Comment (emyl3) [inline README.md:322]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Cloud-Architecture

File: README.md [hunk 22]
```diff
@@ -5,368 +5,17 @@ https://www.simplereport.gov/
...
-![Cloud architecture image](docs/SR_high_level_architecture.png)
-
-
-## Deploy
```
Comment (emyl3) [inline README.md:326]: ⚠️ Moved to https://github.com/CDCgov/prime-simplereport/wiki/Deploy

**Comment by emyl3** [comment]
@emmastephenson Ready for re-review... Added several comments and updated the PR info. Sorry for making you review this twice 😓

**Review by fzhao99 — APPROVED** [review]
Thanks for working on this Elisa! Great work

**Review by emmastephenson — APPROVED** [review]
This looks fantastic, thank you Elisa!! And a big thanks for all the wiki links you included as comments on the PR 🙏

**Review by alismx — APPROVED** [review]
This is great, I'm going to have to update my incoming PRs to follow this pattern!

**Review by rin-skylight — APPROVED** [review]
Thank you for all your hard work on this!

**Review by DanielSass — APPROVED** [review]
Awesome work!
